drivers/can/can_stm32.c

```diff
@@ -419,6 +426,10 @@ static int can_stm32_init(struct device *dev)
 		return ret;
 	}
 
+#ifdef CONFIG_CAN_2
+	filter_can->FMR &= !CAN_FMR_CAN2SB; /* Assign all filters to CAN2 */
```

Comment:
!CAN_FMR_CAN2SB is 0x00 because this is a boolean inversion.
~CAN_FMR_CAN2SB would be the one to use.